src/appleseed.studio/mainwindow/rendering/renderingmanager.cpp

```diff
@@ -490,6 +495,105 @@ void RenderingManager::slot_camera_change_end()
 {
 }
 
+void RenderingManager::slot_material_dropped(const foundation::Vector2d& drop_pos, const std::string& material_name)
```

Comment:
It doesn't seem like this whole code is in the right place in `RenderingManager`. Not sure exactly what would be the best place yet, can you think or something? What about `ObjectInstanceItem` where other material assignments are done?

Reply:
one way would be to create the menu in RenderWidget and connect signal_apply_material to a objectInstanceItem slot slot_apply_material